pkg/services/grpcserver/interceptors/auth.go

```diff
@@ -57,16 +66,18 @@ func (a *Authenticator) tokenAuth(ctx context.Context) (context.Context, error)
 
 	newCtx := purgeHeader(ctx, "authorization")
 
-	_, err = a.getSignedInUser(ctx, token)
+	signedInUser, err := a.getSignedInUser(ctx, token)
 	if err != nil {
 		logger.Warn("request with invalid token", "error", err, "token", token)
 		return ctx, status.Error(codes.Unauthenticated, "invalid token")
 	}
 
+	newCtx = a.contextHandler.SetUser(newCtx, signedInUser)
```

Comment:
we also need to retrieve the permission map for the access control to work, similar to how its done in the HTTP middleware https://github.com/grafana/grafana/blob/bcd7afd1f5201c2438e3e1e5654e9c114af67574/pkg/services/accesscontrol/middleware.go#L119-L141